.github/workflows/publish_pypi.yml

```diff
@@ -88,31 +104,32 @@ jobs:
           python -m pip install cmake wget
           python scripts/install_KLU_Sundials.py
 
-      - name: Build wheels on ${{ matrix.os }}
+      - name: Build wheels on Linux
         run: pipx run cibuildwheel --output-dir wheelhouse
+        if: matrix.os == 'ubuntu-latest'
         env:
           CIBW_ARCHS_LINUX: x86_64
           CIBW_BEFORE_ALL_LINUX: >
             yum -y install openblas-devel lapack-devel &&
             bash scripts/install_sundials.sh 6.0.3 6.5.0
-          CIBW_BEFORE_BUILD_LINUX: >
-            python -m pip install cmake casadi numpy
-          # override; point to casadi install path so that it can be found by the repair command
-          CIBW_REPAIR_WHEEL_COMMAND_LINUX: >
-            LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:$(python -c 'import casadi; print(casadi.__path__[0])')" auditwheel repair -w {dest_dir} {wheel}
+          CIBW_BEFORE_BUILD_LINUX: python -m pip install cmake casadi setuptools wheel
+          CIBW_REPAIR_WHEEL_COMMAND_LINUX: auditwheel repair -w {dest_dir} {wheel}
+          CIBW_TEST_COMMAND: python -c "import pybamm; pybamm.IDAKLUSolver()"
```

Comment:
From the recent discussion on #3193 and #3100, it turns out that checking for `pybamm.have_idaklu()` is not a reliable test and we should try to instantiate an instance of the IDAKLUSolver class instead. This is a very minimal test for now, but we should look at running the integration tests here once we are in a position to migrate to `pytest` soon.